modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php

```diff
@@ -0,0 +1,330 @@
+          array_merge($form['#parents'], ['wrapper', 'edit']),
+        ],
+        '#ajax' => $ajax_definition,
+        '#filter_id' => $filter_id,
+        '#facet_id' => $facet_id,
+        '#executes_submit_callback' => TRUE,
+        '#submit' => [[$this, 'setOperatorSubmit']],
+      ];
+
+      $form['wrapper']['edit'][$filter_id]['cancel_value'] = [
+        '#value' => $this->t('Cancel'),
+        '#type' => 'button',
+        '#op' => 'cancel-contextual-filter',
+        '#name' => static::getFilterType() . '-cancel-' . $filter_id,
+        '#limit_validation_errors' => [
+          array_merge($form['#parents'], [
+            'wrapper',
+            'edit',
+            $filter_id,
+            'cancel_value',
+          ]),
+        ],
+        '#ajax' => $ajax_definition,
+        '#executes_submit_callback' => TRUE,
+        '#submit' => [[$this, 'cancelValueSubmit']],
+      ];
+    }
+
+    return $form;
+  }
+
+  /**
+   * Initialize the form state with the values of the current list source.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   * @param array $configuration
+   *   The current plugin configuration.
+   * @param \Drupal\oe_list_pages\ListSourceInterface $list_source
+   *   The list source.
+   */
+  protected function initializeCurrentContextualFilterValues(FormStateInterface $form_state, array $configuration, ListSourceInterface $list_source): void {
+    // If we have current values for this list source, we can keep them going
+    // forward.
+    $values = static::getCurrentValues($form_state, $list_source);
+    if ($values) {
+      return;
+    }
+
+    // Otherwise, we need to check if the current list source matches the
+    // passed configuration and set the ones from the configuration if they do.
+    // We also check if the values have not been emptied in the current
+    // "session".
+    if ($list_source->getEntityType() === $configuration['entity_type'] && $list_source->getBundle() === $configuration['bundle'] && !static::areCurrentValuesEmpty($form_state, $list_source)) {
+      $values = $configuration['contextual_filters'] ?? [];
+      static::setCurrentValues($form_state, $list_source, $values);
+      return;
+    }
+
+    static::setCurrentValues($form_state, $list_source, []);
+  }
+
+  /**
+   * Ajax request handler for setting a default value for a filters.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   *
+   * @return array
+   *   The form element.
+   */
+  public function setOperatorAjax(array &$form, FormStateInterface $form_state): array {
+    $triggering_element = $form_state->getTriggeringElement();
+    $element = NestedArray::getValue($form, array_slice($triggering_element['#array_parents'], 0, -4));
+    return $element['wrapper'];
+  }
+
+  /**
+   * Submit callback for setting a default value for a filters.
```

Comment:
Similar to above, it should be something like: "Submit callback for setting an operator for a contextual filter."